_posts/current/blog/2022-03-01-how-to-build-flang-on-windows-on-arm.md

```diff
@@ -0,0 +1,94 @@
+
+You should end up with a script that looks something like [this](https://github.com/rovka/f18-llvm-project/blob/flang-woa/flang/examples/build_flang.bat):
+
+```
+1REM You need to modify the paths below:
+2set build_dir=path\to\where\you\want\the\build (must already exist)
+3set clang_root=path\to\where\clang\is\installed
+4set clang_version=x.y.z (should match what’s in %clang_root%)
+5
+6REM Some helper variables.
+7REM Setting CMAKE_CL_SHOWINCLUDES_PREFIX to work around PR27226.
+8set cmake_flags=^
+9  -DCMAKE_BUILD_TYPE=Release ^
+10  -DLLVM_ENABLE_ASSERTIONS=ON ^
+11  -DLLVM_INSTALL_TOOLCHAIN_ONLY=ON ^
+12  -DLLVM_BUILD_LLVM_C_DYLIB=ON ^
+13  -DCMAKE_INSTALL_UCRT_LIBRARIES=ON ^
+14  -DCMAKE_CL_SHOWINCLUDES_PREFIX="Note: including file: " ^
+15  -DLLVM_DEFAULT_TARGET_TRIPLE=aarch64-unknown-windows-msvc ^
+16  -DLLVM_HOST_TRIPLE=aarch64-unknown-windows-msvc ^
+17  -DLLVM_TARGET_ARCH=AArch64 ^
+18  -DCLANG_DEFAULT_LINKER=lld 
+19
+20cd %build_dir%
+21
+22set clang_path=%clang_root%\bin\clang-cl.exe
+23set builtins_path=%clang_root%\lib\clang%clang_version%\lib\windows
+24set builtins_lib=clang_rt.builtins-aarch64.lib
+25
+26set CC=%clang_path%
+27set CXX=%clang_path%
+28
+29REM We enable clang because it is needed by the flang driver.
+30cmake -GNinja %cmake_flags% ^
+31  -DLLVM_ENABLE_PROJECTS="clang;flang;mlir" ^
+32  -DLLVM_TARGETS_TO_BUILD="AArch64" ^
+33  -DCMAKE_C_FLAGS="-fms-compatibility-version=19.20" ^
+34  -DCMAKE_CXX_FLAGS="-fms-compatibility-version=19.20" ^
+35  -DCMAKE_EXE_LINKER_FLAGS="%builtins_path%/%builtins_lib%" ^
+36  ..\llvm-project\llvm || exit /b
+37
+38ninja all || ninja all || ninja all || exit /b
+```
+
+Depending on which revision of flang you’re building, you should now have one or both of f18.exe or flang-new.exe in %build_dir%/bin.
+
+At the time of writing, clang version [13.0.0](https://www.google.com/url?q=https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.0/LLVM-13.0.0-woa64.zip&sa=D&source=docs&ust=1646131480669885&usg=AOvVaw1ILobDb_KMB4Hxid6JOAZa) is known to work for building flang without code generation support.
+However, if you want to build a flang that can generate actual binaries, you need to use custom development branches such as [this one](https://github.com/rovka/f18-llvm-project/commit/6e578e76455eee6ba006ebda911869446c45194c). The latter is based on the community’s [fir-dev branch](https://github.com/flang-compiler/f18-llvm-project/tree/fir-dev), which is currently in the process of being upstreamed, plus some minor changes to get things working on Windows on Arm. 
+
+You can use the steps described above to build even this version of flang, and the resulting executable should be able to compile a simple ‘hello world’ (to link it with the builtins, pass -Xlinker /path/to/clang_rt.builtins-aarch64.lib to flang, just as you did earlier to compile flang itself). 
```

Comment:
If there's a way to do inline code, it would be nice to use it here as well (I'm using ` to demonstrate, not sure if that's what you're supposed to use).
```suggestion
You can use the steps described above to build even this version of flang, and the resulting executable should be able to compile a simple ‘hello world’ (to link it with the builtins, pass `-Xlinker /path/to/clang_rt.builtins-aarch64.lib` to flang, just as you did earlier to compile flang itself). 
```